core/src/toga/app.py

```diff
@@ -98,24 +90,12 @@ def add(self, window: Window) -> None:
             window.app = self.app
 
     def discard(self, window: Window) -> None:
-        """Remove a window from the Window set.
-
-        :param window: The :class:`toga.Window` to remove.
-        """
         if not isinstance(window, Window):
             raise TypeError("Can only discard objects of type toga.Window")
         if window not in self.elements:
             raise ValueError(f"{window!r} is not part of this app")
         self.elements.remove(window)
 
-    def __iadd__(self, window: Window) -> None:
-        self.add(window)
-        return self
-
-    def __isub__(self, other: Window) -> None:
-        self.discard(other)
-        return self
-
```

Comment:
OK, I've restored them and added warnings, since they were never consistent with the standard set interface anyway. 

I could only find one remaining usage in the examples, which I've now removed.